components/org.wso2.carbon.identity.application.authenticator.apple/src/main/java/org/wso2/carbon/identity/application/authenticator/apple/AppleAuthenticator.java

```diff
@@ -145,8 +151,13 @@ protected void initiateAuthenticationRequest(HttpServletRequest request, HttpSer
                 String clientId = authenticatorProperties.get(OIDCAuthenticatorConstants.CLIENT_ID);
                 String authorizationEP = getAuthorizationServerEndpoint(authenticatorProperties);
                 String callbackUrl = getCallbackUrl(authenticatorProperties);
-                String state = getStateParameter(context, authenticatorProperties);
+                if (Boolean.parseBoolean((String) context.getProperty(IS_API_BASED))) {
+                    callbackUrl = resolveCallBackURLForAPIBasedAuthFlow(context);
```

Comment:
can't we have line 153 as the else block of this if?

Reply:
Or we can have `getCallbackUrl` method in this class and define this logic there

Reply:
Addressed in https://github.com/wso2-extensions/identity-outbound-auth-apple/pull/9/commits/9f2bc6ff1d342b6c4ceeec138af629a4634850a5